apps/ccusage/src/_claude-status-api.ts

```diff
@@ -0,0 +1,226 @@
+
+/**
+ * Claude Status API response schema based on actual API response
+ */
+const claudeStatusSchema = v.object({
+	status: v.object({
+		description: v.string(),
+		indicator: v.string(),
+	}),
+	page: v.object({
+		id: v.string(),
+		name: v.string(),
+		url: v.string(),
+		time_zone: v.string(),
+		updated_at: v.string(),
+	}),
+});
+
+export type ClaudeStatus = v.InferInput<typeof claudeStatusSchema>;
+
+/**
+ * Status indicator types based on common status page indicators
+ */
+export type StatusIndicator = 'none' | 'minor' | 'major' | 'critical';
+
+/**
+ * Get the appropriate color formatter for Claude status
+ * @param indicator - Status indicator from API
+ * @param description - Status description for fallback detection
+ * @param enableColors - Whether to enable colors (default: auto-detect)
+ * @returns Color formatter function
+ */
+export function getStatusColor(
+	indicator: string,
+	description: string,
+	enableColors?: boolean,
+): Formatter {
+	// Determine if colors should be enabled
+	const shouldColor = enableColors ?? pc.isColorSupported;
+
+	if (!shouldColor) {
+		return (text: string | number | null | undefined) => String(text ?? '');
+	}
+
+	// Primary check: indicator-based coloring
+	if (indicator === 'none' || description.toLowerCase().includes('operational')) {
+		return pc.green;
+	}
+	else if (indicator === 'minor' || description.toLowerCase().includes('degraded')) {
+		return pc.yellow;
+	}
+	else if (indicator === 'major' || indicator === 'critical' || description.toLowerCase().includes('outage')) {
+		return pc.red;
+	}
+
+	// Default: no coloring for unknown status
+	return (text: string | number | null | undefined) => String(text ?? '');
+}
+
+/**
+ * Fetch Claude status from status.claude.com API
+ * @returns Result containing Claude status data or error
+ */
+export async function fetchClaudeStatus(): Result.ResultAsync<ClaudeStatus, Error> {
+	const result = Result.try({
+		try: async () => {
+			const response = await fetch('https://status.claude.com/api/v2/status.json');
+
+			if (!response.ok) {
+				throw new Error(`Failed to fetch Claude status: ${response.status} ${response.statusText}`);
+			}
+
+			const data: unknown = await response.json();
+
+			// Validate response data using safeParse
+			const parseResult = v.safeParse(claudeStatusSchema, data);
+			if (!parseResult.success) {
+				throw new Error(`Invalid API response format: ${parseResult.issues.map(issue => issue.message).join(', ')}`);
+			}
+
+			return parseResult.output;
+		},
+		catch: (error: unknown) => error instanceof Error ? error : new Error(String(error)),
+	});
+
+	return result();
+}
+
+if (import.meta.vitest != null) {
+	const { describe, it, expect } = import.meta.vitest;
+
+	describe('fetchClaudeStatus', () => {
+		it('should fetch and parse actual API response', async () => {
+			const result = await fetchClaudeStatus();
+
+			// Only run test if API is available
+			if (Result.isSuccess(result)) {
```

Comment:
These conditionals exist because we're making actual HTTP requests without mocking `fetch`. The tests would fail unpredictably based on network/API availability otherwise.
Should I implement fetch mocking with `vi.mock()` to eliminate these conditionals and make the tests deterministic? Happy to refactor if that's preferred.